content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md

```diff
@@ -0,0 +1,434 @@
+{{< /note >}}
+
+{{< tabs name="tab_with_md" >}}
+{{% tab name="Self Managed" %}}
+
+### Self managed
+Switch to the **Managed service** tab if you are connecting to Elasticsearch Service in Elastic Cloud.
+
+### Set the credentials
+There are four files to edit to create a k8s secret when you are connecting to self managed Elasticsearch and Kibana (self managed is effectively anything other than the managed Elasticsearch Service in Elastic Cloud).  The files are:
+
+1. ELASTICSEARCH_HOSTS
+1. ELASTICSEARCH_PASSWORD
+1. ELASTICSEARCH_USERNAME
+1. KIBANA_HOST
+
+Set these with the information for your Elasticsearch cluster and your Kibana host.  Here are some examples
+
+#### `ELASTICSEARCH_HOSTS`
+1. A nodeGroup from the Elastic Elasticseach Helm Chart:
+
+    ```shell
+    ["http://elasticsearch-master.default.svc.cluster.local:9200"]
+    ```
+1. A single Elasticsearch node running on a Mac where your Beats are running in Docker for Mac:
+
+    ```shell
+    ["http://host.docker.internal:9200"]
+    ```
+1. Two Elasticsearch nodes running in VMs or on physical hardware:
+
+    ```shell
+    ["http://host1.example.com:9200", "http://host2.example.com:9200"]
+    ```
+Edit `ELASTICSEARCH_HOSTS`
+```shell
+vi ELASTICSEARCH_HOSTS
+```
+
+#### `ELASTICSEARCH_PASSWORD`
+Just the password, no whitespace or quotes:
+
+    changeme
+
+Edit `ELASTICSEARCH_PASSWORD`
+```shell
+vi ELASTICSEARCH_PASSWORD
+```
+
+#### `ELASTICSEARCH_USERNAME`
+Just the username, no whitespace or quotes:
+
+    elastic
+
+Edit `ELASTICSEARCH_USERNAME`
+```shell
+vi ELASTICSEARCH_USERNAME
+```
+
+#### `KIBANA_HOST`
+
+1. The Kibana instance from the Elastic Kibana Helm Chart.  The subdomain `default` refers to the default namespace.  If you have deployed the Helm Chart using a different namespace, then your subdomain will be different:
+
+    ```shell
+    "kibana-kibana.default.svc.cluster.local:5601"
+    ```
+1. A Kibana instance running on a Mac where your Beats are running in Docker for Mac:
+
+    ```shell
+    "host.docker.internal:5601"
+    ```
+1. Two Elasticsearch nodes running in VMs or on physical hardware:
+
+    ```shell
+    "host1.example.com:5601"
+    ```
+Edit `KIBANA_HOST`
+```shell
+vi KIBANA_HOST
+```
+
+### Create a Kubernetes secret
+This command creates a secret in the Kubernetes system level namespace (kube-system) based on the files you just edited:
+
+    kubectl create secret generic dynamic-logging \
+      --from-file=./ELASTICSEARCH_HOSTS \
+      --from-file=./ELASTICSEARCH_PASSWORD \
+      --from-file=./ELASTICSEARCH_USERNAME \
+      --from-file=./KIBANA_HOST \
+      --namespace=kube-system
+
+{{% /tab %}}
+{{% tab name="Managed service" %}}
+
+## Managed service
+This tab is for Elasticsearch Service in Elastic Cloud only, if you have already created a secret for a self managed Elasticsearch and Kibana deployment, then skip this continue with [Deploy the Beats](#deploy-the-beats).
```

Comment:
I'm not clear on what this line is trying to say.

Reply:
that is a mess :)

Reply:
I wanted to make sure that people do not run through the instructions on both tabs, so at the top of the first tab I direct them to the second tab if they are in the wrong place, and at the top of the second tab I direct them to the next section of the doc if they have already created the secret in the first tab.

Do you think I am going overboard in trying to prevent errors?